pkg/workloads/console/controller.go

```diff
@@ -505,19 +506,28 @@ func (r *reconciler) generateStatusAndAuditEvents(statusCtx consoleStatusContext
 		logger.Log("event", ConsoleStarted, "msg", "Console started")
 	}
 
-	// Console phase from Running to Stopped
+	// Console phase from Running to Stopped, with a new CompletionTime: the job
+	// completed successfully
 	if newStatus.CompletionTime != r.console.Status.CompletionTime && !statusCtx.Job.Status.StartTime.IsZero() {
 		duration := statusCtx.Job.Status.CompletionTime.Sub(statusCtx.Job.Status.StartTime.Time).Seconds()
 		logger.Log("event", ConsoleEnded, "msg", "Console ended", "duration", duration)
 	}
 
-	// Console phase from Running to Stopped without CompletionTime (expire)
+	// Console phase from Running to Stopped without CompletionTime: the job's
+	// activeDeadlineSeconds was reached, the job was marked as failed and the
+	// pod deleted.
 	if r.console.Running() &&
 		newStatus.CompletionTime == nil && newStatus.Phase == workloadsv1alpha1.ConsoleStopped {
 		duration := r.console.Status.ExpiryTime.Sub(statusCtx.Job.Status.StartTime.Time).Seconds()
-		logger.Log("event", ConsoleEnded, "msg", "Console ended after reaching expiration time", "duration", duration)
+		logger.Log("event", ConsoleEnded, "msg", "Console ended due to expiration", "duration", duration)
+	}
+
+	// Console was in PendingAuthorisation phase, but is about to be deleted.
+	if r.console.PendingAuthorisation() && r.console.EligibleForGC() {
```

Comment:
I put this here, rather than amending the logger in the block where we delete the console if it's eligible for GC because it became clear that these are two different cases: Here we're generating an audit log for human consumption (and therefore it makes sense for it to be co-located with the rest of the audit logs), and in the other block we're primarily creating a Kubernetes event, which happens to be via the logger.